catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,502 @@
+        if (ops == null) {
+            LOGGER.warn("There was no operations staged for historian transaction key [{}]",
+                    historianTransactionKey);
+            return null;
+        }
+
+        ArrayList<Exception> exceptions = new ArrayList<>();
+        for (Callable<Boolean> op : ops) {
+            try {
+                op.call();
+            } catch (Exception e) {
+                exceptions.add(e);
+            }
+        }
+        return exceptions;
+    }
+
+    /**
+     * Removes a set of staged operations.
+     * </p>
+     * Calling multiple times or on a non-existent key has no effect.
+     *
+     * @param historianTransactionKey The key associated with the set of operations to upstage.
+     */
+    public void rollback(String historianTransactionKey) {
+        staged.remove(historianTransactionKey);
+    }
+
+    public boolean isHistoryEnabled() {
+        return historyEnabled;
+    }
+
+    public void setHistoryEnabled(boolean historyEnabled) {
+        this.historyEnabled = historyEnabled;
+    }
+
+    public List<StorageProvider> getStorageProviders() {
+        return storageProviders;
+    }
+
+    public void setStorageProviders(List<StorageProvider> storageProviders) {
+        this.storageProviders = storageProviders;
+    }
+
+    public List<CatalogProvider> getCatalogProviders() {
+        return catalogProviders;
+    }
+
+    public void setCatalogProviders(List<CatalogProvider> catalogProviders) {
+        this.catalogProviders = catalogProviders;
+    }
+
+    private StorageProvider storageProvider() {
+        return storageProviders.stream()
+                .findFirst()
+                .orElse(null);
+    }
+
+    private CatalogProvider catalogProvider() {
+        return catalogProviders.stream()
+                .findFirst()
+                .orElse(null);
+    }
+
+    private CreateResponse getVersionedMetacards(List<Metacard> metacards,
+            final MetacardVersion.Action action)
+            throws SourceUnavailableException, IngestException {
+        final List<Metacard> versionedMetacards = metacards.stream()
+                .filter(MetacardVersion::isNotVersion)
+                .map(metacard -> new MetacardVersion(metacard, action, SecurityUtils.getSubject()))
+                .collect(Collectors.toList());
+
+        if (versionedMetacards.isEmpty()) {
+            return null;
+        }
+
+        return getSystemSubject().execute(() -> catalogProvider().create(new CreateRequestImpl(
+                versionedMetacards)));
+    }
+
+    private List<ContentItem> getVersionedContent(Metacard root, List<ContentItem> contentItems,
+            MetacardVersion.Action versionAction, Map<String, Path> tmpContentPaths) {
+        String id = root.getId();
+        MetacardVersion rootVersion = new MetacardVersion(root,
+                versionAction,
+                SecurityUtils.getSubject());
```

Comment:
The Historian is not a bean or bundle so I could not inject a supplier. The Subject is a runtime mechanism so it is extracted to the constructor so that the calling class can choose how to get the Subject. Unless you mean something different than I am understanding